public/update-template.js

```diff
@@ -1,60 +1,77 @@
 $(document).ready(() => {
   const urlParams = new URLSearchParams(window.location.search);
-  const templateName = urlParams.get('name');
+  const templateName = urlParams.get("name");
 
   if (!templateName) {
-    window.location.href = '/'; //something went wrong
+    window.location.href = "/"; //something went wrong
   }
 
-  window.codeMirrorEditor = window.CodeMirror.fromTextArea(document.querySelector('#codeMirror'), {
-    mode: "htmlmixed",
-    lineNumbers: true
-  });
+  window.codeMirrorEditor = window.CodeMirror.fromTextArea(
+    document.querySelector("#codeMirror"),
+    {
+      mode: "htmlmixed",
+      lineNumbers: true,
+    }
+  );
 
-  $.get(`/get-template/${templateName}?region=${localStorage.getItem('region')}`, function (response) {
-    $('#templateName').val(response.data.TemplateName);
-    $('#templateSubject').val(response.data.SubjectPart);
-    $('#templateText').val(response.data.TextPart);
+  window.fillVarsCodeMirrorEditor = window.CodeMirror.fromTextArea(
+    document.querySelector("#fillVarsCodeMirror"),
+    {
+      mode: "application/json",
+      lineNumbers: true,
+    }
+  );
 
-    window.codeMirrorEditor.setValue(response.data.HtmlPart ? response.data.HtmlPart : "");
+  setInterval(() => {
```

Comment:
Oh wow, the editor will be redrawn every 100ms?!

Reply:
I fixed this, now it will only refresh when opened. There is one little downside of this though it is better than refreshing every 100ms, Thank you for noting that.

Reply:
Does not require page refresh, the downside is that the variables will flash for a short delay ( about 200ms ) when the popup is open.